lib/vdsm/storage/managedvolume.py

```diff
@@ -297,3 +304,28 @@ def _remove_udev_rule(vol_id):
     except Exception:
         log.exception(
             "Failed to remove udev rule for volume %s", vol_id)
+
+
+def _add_run_link(vol_id, path):
+    _create_managedvolume_link_dir()
+    run_path = os.path.join(VOLUME_LINK_DIR, vol_id)
+    os.symlink(path, run_path)
+
+    return run_path
+
+
+def _remove_run_link(vol_id):
+    try:
+        os.symlink(os.path.join(VOLUME_LINK_DIR, vol_id))
+    except:
+        log.exception("Failed to remove run link for volume %s", vol_id)
+
+
+def _create_managedvolume_link_dir():
+    try:
+        os.mkdir(VOLUME_LINK_DIR)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
```

Comment:
This was the best way in python 2, but in python 3 we can use:

    try:
        os.mkdir(path)
    except FileExistsError:
        pass